h2o-algos/src/main/java/hex/ensemble/StackedEnsemble.java

```diff
@@ -123,7 +123,9 @@ static void addModelPredictionsToLevelOneFrame(Model aModel, Frame aModelsPredic
     if (aModel._output.isBinomialClassifier()) {
       // GLM uses a different column name than the other algos
       Vec preds = aModelsPredictions.vec(2); // Predictions column names have been changed...
-      levelOneFrame.add(aModel._key.toString(), preds);
+      synchronized (levelOneFrame) {
```

Comment:
I would be happier if this method didn't use synchronization. The semantics can be changed to instead return a Frame wrapping the columns that would be later stitched together to "levelOneFrame".

1. This method will return new columns
2. Columns will be cbinded together here: https://github.com/h2oai/h2o-3/pull/4524/files#diff-91d255cd89e95ed4a240332df7d9002bR150

WDYT? cbind is a cheap operation

Reply:
@michalkurka Thank you for the comment!

I tried but didn't find a reasonable way how to collect results from `Futures` so I looked into using `LocalMR`. I came up with the following. It doesn't use `synchronized` but I kept the semantics of the method the same and call it sequentially since it's not the bottleneck (the method takes some columns, renames them, and cbinds them) and it might complicate cleanup.

Performance-wise, I didn't see any significant difference to the previous solution on my mini benchmark.

I didn't commit this change since I am not sure with the use of `LocalMR` and its `nthreads` argument which is often used the way I did - as a way to call `MrFun::map(i) for i in 0..nthreads`. Is that OK? Is that any better than the previous solution?

```java
    Model[] usefulBaseModels = Stream.of(_parms._base_models)
            .filter(this::isUsefulBaseModel)
            .map(Key::get)
            .toArray(Model[]::new);

    Frame[] baseModelPredictions = new Frame[usefulBaseModels.length];

    // Run scoring of base models in parallel
    H2O.submitTask(new LocalMR(new MrFun() {
      @Override
      protected void map(int id) {
        baseModelPredictions[id] = usefulBaseModels[id].score(
                fr,
                "preds_base_" + seKey + usefulBaseModels[id]._key + fr._key,
                j,
                false
        );
      }
    }, usefulBaseModels.length)).join();

    for (int i = 0; i < usefulBaseModels.length; i++) {
      StackedEnsemble.addModelPredictionsToLevelOneFrame(usefulBaseModels[i], baseModelPredictions[i], levelOneFrame);
      DKV.remove(baseModelPredictions[i]._key);
      Frame.deleteTempFrameAndItsNonSharedVecs(baseModelPredictions[i], levelOneFrame);
    }
```